micro-rdk/src/native/entry.rs

```diff
@@ -145,132 +153,124 @@ pub async fn serve_web_inner(
     srv.serve(robot).await;
 }
 
+async fn validate_robot_credentials(
+    exec: NativeExecutor,
+    robot_creds: &RobotCredentials,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let app_config = AppClientConfig::new(
+        robot_creds.robot_secret().to_owned(),
+        robot_creds.robot_id().to_owned(),
+        "".to_owned(),
+    );
+    let conn = NativeStream::TLSStream(Box::new(
+        NativeTls::new_client().open_ssl_context(None).await?,
+    ));
+    let client = GrpcClient::new(conn, exec.clone(), "https://app.viam.com:443").await?;
+    let builder = AppClientBuilder::new(Box::new(client), app_config.clone());
+
+    let _client = builder.build().await?;
+    Ok(())
+}
+
 #[cfg(feature = "provisioning")]
-async fn serve_provisioning_async<S>(
-    ip: Ipv4Addr,
+async fn serve_async_with_external_network<S>(
```

Comment:
This function and the one below are named `_with_external_network`, but they accept `impl Network` which isn't necessarily an ExternalNetwork.